pkg/storage/filepath/jsonfile_rest.go

```diff
@@ -146,6 +150,13 @@ func (f *filepathREST) Create(
 	createValidation rest.ValidateObjectFunc,
 	options *metav1.CreateOptions,
 ) (runtime.Object, error) {
+
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, err
+	}
+	rest.FillObjectMetaSystemFields(accessor)
```

Comment:
this part is the main change. it assigns the CreationTimestamp and UID. Not totally sure where this was handled before, but now it's expected to be handled before BeforeCreate()